lib/ansible/parsing/yaml/dumper.py

```diff
@@ -59,6 +60,10 @@ def represent_binary(self, data):
         return yaml.representer.SafeRepresenter.represent_str(self, binary_type(data))
 
 
+def represent_undefined(self, data):
+    return bool(data)
```

Comment:
Just to clarify why this works, jinja sets `__bool__` on `StrictUndefined` to a method that will raise an exception on use:

https://github.com/pallets/jinja/blob/a2e83d07a30e1d6f1242993e694f8fdfad1d23f2/src/jinja2/runtime.py#L1093

So this just triggers `Undefined._fail_with_undefined_error` so we get the error raised.

Reply:
Thanks for the information.

Reply:
We do this (or similar using `str()`) in at least 2 other places and document why it works at each invocation. We could add a method like `error_if_undefined` that would better explain what is the intent. Just a thought, no need to do this now.

Reply:
i would add comment with what sivel mentioned above to avoid scar marks in head of next person to look at this code